ui/src/dataExplorer/components/DashboardsDropdown.tsx

```diff
@@ -55,13 +56,18 @@ class DashboardsDropdown extends PureComponent<Props> {
   }
 
   private get dropdownLabel(): string {
-    const {dashboards, selectedIDs} = this.props
+    const {dashboards, selectedIDs, newDashboardName} = this.props
 
     if (!selectedIDs.length) {
       return 'Choose at least 1 dashboard'
     }
 
-    return dashboards
+    const dashboardsWithNew = [
+      ...dashboards,
+      {...DashboardTemplate, name: newDashboardName},
+    ]
+
+    return dashboardsWithNew
```

Comment:
could this have been:
```suggestion
    return [
      ...dashboards,
      {...DashboardTemplate, name: newDashboardName},
    ]
```
?

Looks good either way

Reply:
that would be better thanks!

Reply:
that is equivalent but I find that a bit harder to read.. ? thanks!